src/lib.rs

```diff
@@ -463,6 +463,17 @@ impl usb_device::bus::UsbBus for UsbBus {
             PollResult::None
         })
     }
+
+    fn force_reset(&self) -> usb_device::Result<()> {
+        interrupt::free(|cs| {
+            self.usb
+                .borrow(cs)
+                .udcon
+                .modify(|_, w| w.detach().set_bit());
+
```

Comment:
Should this also clear the detach bit? From the datasheet:

> It is possible to re-enumerate a device, simply by setting and clearing the DETACH bit (but firmware must take
in account a debouncing delay of some milliseconds).

The only place where the detach bit is cleared in this driver is in `enable()`. However the [trait docs](https://docs.rs/usb-device/latest/usb_device/bus/trait.UsbBus.html#method.force_reset) don't specify any steps to re-initialize the device after  force reset, so I think this should be a self-contained detach and attach sequence.

Reply:
FWIW, [that's what the nRF USB driver does](https://docs.rs/nrf-usbd/0.2.0/src/nrf_usbd/usbd.rs.html#733-747)

Reply:
I'll probably take care of this myself. If we don't want to depend on `avr_hal::Delay`, then we need to write a cycle delay routine (like what the nRF driver has).

For now I'll put it in this crate but I'll also submit a pull request to `avr-device`. I think that would be useful for them to have.

Reply:
> I'll probably take care of this myself. If we don't want to depend on avr_hal::Delay, then we need to write a cycle delay routine (like what the nRF driver has) to delay for at least 1 millisecond (16000 cycles at 16 MHz)

Sounds good, I mostly need this for a project I'm working on, and the upstream for that project just does this simple `detach`. 

AFAICT, the `enable` call is only made during `UsbDevice` creation, since the rest of the trait API looks to intentionally make it impossible to directly call `enable`.

There does seem to be some verbiage about the device resetting all device state after a reset, so maybe it is intended for users to simply re-create a `UsbDevice` after a `force_reset`, IDK